compat/signal.h

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 1996-2025 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_COMPAT_SIGNAL_H
+#define SQUID_COMPAT_SIGNAL_H
+
+#if !defined(SIGHUP)
+#define SIGHUP  1   /* hangup */
+#endif
+#if !defined(SIGKILL)
+#define SIGKILL 9   /* kill (cannot be caught or ignored) */
+#endif
+#if !defined(SIGBUS)
+#define SIGBUS  10  /* bus error */
+#endif
+#if !defined(SIGPIPE)
+#define SIGPIPE 13  /* write on a pipe with no one to read it */
+#endif
+#if !defined(SIGALRM)
+#define SIGALRM 14  /* real-time timer expired */
+#endif
+#if !defined(SIGCHLD)
+#define SIGCHLD 20  /* to parent on child stop or exit */
+#endif
+#if !defined(SIGUSR1)
+#define SIGUSR1 30  /* user defined signal 1 */
+#endif
+#if !defined(SIGUSR2)
+#define SIGUSR2 31  /* user defined signal 2 */
+#endif
+
+/// POSIX kill(2) equivalent
+int xkill(pid_t pid, int sig);
+
+/// true if pid can be sent a signal (no signal is sent)
+inline bool
+IsPidValid(pid_t pid);
+
```

Comment:
This is not part of the POSIX API. We should be using that API instead of local wrappers.
```suggestion
```

Reply:
I agree that `compat/signal.h` should not provide `IsPidValid()` API.

Instead, `xkill()` implementation for Windows may (and probably should!) use a static helper function, hopefully with a more revealing name, to implement `xkill(signal=0)` support).

Old higher-level code should continue to use `ProcessIsRunning()` and raw `xkill(signal=0)` calls.